### PR TITLE
app/vmselect/promql: check for ranged vectors in aggr funcs if implicit conversions are disabled

### PR DESCRIPTION
**Pull request by hagen1778**
Check for ranged vector arguments in aggregate expressions when `-search.disableImplicitConversion` or `-search.logImplicitConversion` are enabled.
 For example, `sum(up[5m])` will fail to execute if these flags are set.

### Describe Your Changes

Please provide a brief description of the changes you made. Be as specific as possible to help others understand the purpose and impact of your modifications.

### Checklist

The following checks are **mandatory**:

- [*] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

### CODE REVIEW

**Review by f41gh7 — APPROVED** [review]
LGTM

**Comment by Haleygo** [comment]
What about transform functions like `absent(up[5m])`, should we disable they here or in followup? Since they're also wrong expressions.

File: app/vmselect/promql/exec.go [hunk 1]
```diff
@@ -452,6 +453,11 @@ func isSubQueryComplete(e metricsql.Expr, isSubExpr bool) bool {
 			return false
 		}
 		for _, arg := range exp.Args {
+			if re, ok := arg.(*metricsql.RollupExpr); ok {
+				if re.Window != nil {
```
Comment (Haleygo) [inline app/vmselect/promql/exec.go:457]: Should we also check for wrong expr like `sum(up[:5m])`?

Comment (hagen1778) [reply app/vmselect/promql/exec.go:462]: Good point! Please see updates in https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6450/commits/296902003ac1cd8b5220f6832d3f9b52705b05c9

Note, it now forbids use of `rate(metric)` when no ranged vector is specified - it is what you suggested from the very beginning. I think you were right to restrict this.

File: app/vmselect/promql/exec_test.go [hunk 2]
```diff
@@ -9495,17 +9496,25 @@ WITH (
...
+func TestNoImplicitConversionRequiredFalse(t *testing.T) {
 	f := func(q string) {
 		t.Helper()
 		e, err := metricsql.Parse(q)
 		if err != nil {
 			t.Fatal(err)
 		}
-		if isSubQueryComplete(e, false) {
+		if noImplicitConversionRequired(e, false) {
 			t.Fatalf("expect to detect incomplete subquery: %s", e.AppendString(nil))
```
Comment (Haleygo) [inline app/vmselect/promql/exec_test.go:9507]: nit, might change this error message as well

**Review by Haleygo — APPROVED** [review]
LGTM

**Comment by valyala** [comment]
Please see the follow-up commit 61d794c5e759025366d39d477d8a2ad63e0cacf5

**Pull request merged**